applications/FluidDynamicsApplication/custom_utilities/fluid_characteristic_numbers_utilities.h

```diff
@@ -0,0 +1,190 @@
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+
+// Application includes
+
+
+namespace Kratos
+{
+
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) FluidCharacteristicNumbersUtilities
+{
+public:
+
+	///@name Type Definitions
+	///@{
+
+	/// Pointer definition of FluidCharacteristicNumbersUtilities
+	KRATOS_CLASS_POINTER_DEFINITION(FluidCharacteristicNumbersUtilities);
+
+    /// Function type for the element size calculator function
+    typedef std::function<double(const Geometry<Node<3>>&)> ElementSizeFunctionType;
+
+	///@}
+	///@name Life Cycle
+	///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Calculate each element's CFL for the current time step and provided model part
+     * The elemental CFL is stored in the CFL_NUMBER elemental variable
+     * To visualize in the post-process file, remember to print CFL_NUMBER as a Gauss point result
+     */
+    static void CalculateLocalCFL(ModelPart& rModelPart);
+
+    /**
+     * @brief Calulate element CFL number
+     * For the given element, this method calculates the CFL number
+     * @param rElement Element to calculate the CFL number
+     * @param rGeometryInfo Auxiliary geometry data container
+     * @param Dt Current time increment
+     * @return double The element CFL number
+     */
+    static double CalculateElementCFL(
+        const Element &rElement,
+        const ElementSizeFunctionType& rElementSizeCalculator,
+        const double Dt);
+
+    /**
+     * @brief Calculate the element Prandtl number
+     * For the given element, this method calculates the Prandtl number
+     * @tparam ConsiderArtificialMagnitudes Template parameter specifying if the artificial values are considered 
+     * @param rElement Element to calculate the Prandtl number
+     * @return double The element Prandtl number
+     */
+    template<bool ConsiderArtificialMagnitudes>
+    static double CalculateElementPrandtlNumber(const Element& rElement);
+
+    /**
+     * @brief Calculate the elemental Peclet numbers
+     * For the given element, this method calculates the Peclet number considering both the
+     * dynamic viscosity and the shear conductivity 
+     * @tparam ConsiderArtificialMagnitudes Template parameter specifying if the artificial values are considered
+     * @tparam DensityIsNodal Template parameter specifying if the density is nodally stored
+     * @param rElement Element to calculate the Peclet number
+     * @return std::tuple<double,double> Tuple containing the viscosity (first position) and conductivity (second position) Peclet numbers
+     */
+    template<bool ConsiderArtificialMagnitudes, bool DensityIsNodal>
+    static std::tuple<double,double> CalculateElementPecletNumbers(
```

Comment:
you should probably include the `tuble` header for completeness

Reply:
Done!